snippets/puppet_setup.erb

```diff
@@ -58,4 +58,26 @@ fi
 <% end -%>
 <% end -%>
 
-<%= bin_path %>/puppet agent --config <%= etc_path %>/puppet.conf --onetime --tags no_such_tag <%= @host.puppetmaster.blank? ? '' : "--server #{@host.puppetmaster}" %> --no-daemonize
+<% if @host.param_true?('run-puppet-in-installer') -%>
+<% if @host.operatingsystem.name == 'Ubuntu' and @host.operatingsystem.major.to_i >= 15 -%>
```

Comment:
Sure. It probably could be extended for other OSes as well, but I didn't test those and therefore didn't want to include a workaround based on guesswork.